.jenkins/build_container.JenkinsFile

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/groovy
+def LABEL = "build-jenkins-casc-" + UUID.randomUUID().toString().take(8)
+pipeline {
+    parameters {
+        choice(name: 'TAG', choices: ['git-commit','git-tag'],  description: 'Which info to use in container tag?')
```

Comment:
I'm a little puzzled / the architectural choice is not totally clear to me.
Do I interpret correctly that you want to allow us to choose, whether we use the commit as label or the tag? Side question: why not applying both (tag only if there's a tag for this commit, of course)?

You can have jenkins jobs (auto-created by the Pipeline Multibranch / Github Branch Source) plugins for both branches as well as tags.
So you want to allow us to push containers based on a branch (should we actually restrict pushing to master branch?) as well as based on a tag that is associated with the latest commit of that branch?
I would have to look it up, but I remember BA has some pipeline(s) which act only based on tags. This would be also an option.

But I find it a little weird to select in a branch-based pipeline (which I assume you define in Job DSL) to then deal with an existing tag. That alternative architectural choice would make sense to me, if we tag automatically after a successful build. But I think that's not the best one here (because we want to tag with the Jenkins version).

I hope this was not confusing. My suggestion would be (because we cannot test this easily locally, right):

- start the first iteration with using the current commit as label, then iterate on the pipeline
- configure in Job DSL that it generates jobs for tags
- update the pipeline with a `when { tag ...` condition to then apply that label
- think about restricting pushes to ECR when we're not working on tagged source code (which.. the more I think about it, might not be absolutely needed).

Advantage would be here: we can let the pipeline run for any branch and see if the container builds. Then we would manually tag (I guess), a new job for the tag appears, it is built (hopefully automatically) and pushes the container again (*).

(*) of course, this violates the rule to build it only once. So way better for the tag-based job would be to add additional labels to the same image.. dear..

Reply:
I think I understood :D
Well, let's keep only the commit hash, for now, then I can work on something to append with the commit tag.